src/Framework/AspNetCoreAnalyzers/src/CodeFixes/Dependencies/ThisAndExtensionMethod.cs

```diff
@@ -5,4 +5,20 @@
 
 namespace Microsoft.AspNetCore.Analyzers;
 
-internal record struct ThisAndExtensionMethod(ITypeSymbol thisType, string extensionMethod);
+internal readonly struct ThisAndExtensionMethod(ITypeSymbol thisType, string extensionMethod)
+{
+    public ITypeSymbol ThisType { get; } = thisType;
+    public string ExtensionMethod { get; } = extensionMethod;
+
+    public override bool Equals(object obj)
+    {
+        return obj is ThisAndExtensionMethod other &&
+            SymbolEqualityComparer.Default.Equals(ThisType, other.ThisType) &&
+            ExtensionMethod == other.ExtensionMethod;
+    }
+
+    public override int GetHashCode()
+    {
+        return HashCode.Combine(ThisType, ExtensionMethod);
```

Comment:
This should use `SymbolEqualityComparer.Default.GetHashCode(ThisType)` to match the above